src/main/java/com/google/graphgeckos/dashboard/datatypes/BuildBotData.java

```diff
@@ -121,4 +121,18 @@ public void setTimestamp(@NonNull Timestamp timestamp) {
     this.timestamp = timestamp;
   }
 
+  /**
+   * Checks if all non-transient fields are equal between BuildBotData instances.
+   */
+  @Override
+  public boolean equals(Object o) {
```

Comment:
Consider adding ```if (o == this) return true```

Reply:
Why? This is unlikely to happen, and if this happens it's covered by the last if. The value we get from adding this special case it's overshadowed by what performance we lose every time we check for this.

Reply:
Ok!